rust/otap-dataflow/crates/config/src/pipeline_group.rs

```diff
@@ -13,25 +13,42 @@ use std::collections::HashMap;
 /// Contains group-specific settings and all its pipelines.
 #[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
 pub struct PipelineGroupConfig {
-    /// An optional description of the pipeline group.
-    pub description: Option<Description>,
-
-    /// Settings that apply to this pipeline group only.
-    pub settings: PipelineGroupSettings,
-
     /// All pipelines belonging to this pipeline group, keyed by pipeline ID.
     pub pipelines: HashMap<PipelineId, PipelineConfig>,
-    // ToDo: Add resource quota support.
-}
 
-/// Pipeline group-specific settings.
-#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
-pub struct PipelineGroupSettings {
-    /// If false, the pipeline group is disabled and cannot process data.
-    pub enabled: bool,
+    /// Quota for the pipeline group.
+    #[serde(default)]
+    pub quota: Quota,
 }
 
 impl PipelineGroupConfig {
+    /// Creates a new empty pipeline group configuration.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            pipelines: HashMap::new(),
+            quota: Default::default(),
+        }
+    }
+
+    /// Sets the quota for the pipeline group.
+    pub fn set_quota(&mut self, quota: Quota) {
+        self.quota = quota;
+    }
+
+    /// Adds a pipeline to the pipeline group.
+    pub fn add_pipeline(
+        &mut self,
+        pipeline_id: PipelineId,
+        pipeline: PipelineConfig,
+    ) -> Result<(), Error> {
+        let prev_pipeline_grp = self.pipelines.insert(pipeline_id.clone(), pipeline);
+        if prev_pipeline_grp.is_some() {
+            return Err(Error::DuplicatePipeline { pipeline_id });
+        }
```

Comment:
nit: should this check that the pipelines already has the pipeline, and return an error before doing the insert? Otherwise, caller might receive the error and expect that the new pipeline was not added to `PipelineGroupConfig`, when in fact it was

Reply:
Fixed